CMakeLists.txt

```diff
@@ -75,6 +75,56 @@ include(InstallRequiredSystemLibraries)
 option(BUILD_SHARED_LIBS "Build shared libraries" ON)
 set(BUILD_VERSION "0.0.0" CACHE STRING "Library version (for both libbsoncxx and libmongocxx)")
 
+# Allow the user to decide whether to also build static libraries
+option(BUILD_SHARED_AND_STATIC_LIBS "Build static libraries" OFF)
+
+# Allow the user to decide whether to use shared libraries or static libraries
+# for the mongo-c-driver
+option(BUILD_SHARED_LIBS_WITH_STATIC_MONGOC
+       "Use static mongo-c-driver libraries with shared mongo-cxx-driver libraries"
+        OFF
+)
+
+# Disambiguate our options into clear flags
+if(BUILD_SHARED_LIBS)
```

Comment:
So I added two external options. However, I wanted to make sure I maintained existing behavior. So I also added these six INTERNAL variables to explicitly turn on and off features. If this is too messy, I can definitely hide it in a cmake module.

Reply:
I like it – the six internal variables clarify the logic.